src/frontend/Assets/Scripts/PythonRunner.cs

```diff
@@ -0,0 +1,86 @@
+using System.Diagnostics;
+using System.IO;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+
+public class PythonScriptRunner : MonoBehaviour
+{
+    // Serialize input parameters as private to make them visible in the Unity Editor
+    [SerializeField] private double inclineAngle = 0.0;
+
+    // Serialize reference objects
+    [SerializeField] private Button runButton;
+    [SerializeField] private TMP_InputField inclineAngleInput;
+
+    private void Start()
+    {
+        // Add a listener to the run button
+        runButton.onClick.AddListener(RunPython);
+
+        // Add listeners for parameter inputs
+        inclineAngleInput.onValueChanged.AddListener(UpdateInclineAngle);
+
+    }
+
+    private void UpdateInclineAngle(string value)
+    {
+        // Try to parse the input value to a double
+        if (double.TryParse(value, out double angle))
+        {
+            inclineAngle = angle;
+        } else {
+            UnityEngine.Debug.LogError("Invalid input for incline angle: " + value);
+        }
+    }
+
+    public void RunPython()
+    {   
+        // Resolve the path to the python environment and main file of the python script
+        string pythonPath = Path.GetFullPath(Path.Combine(Application.dataPath, "../../../.venv/Scripts/python.exe"));
+        string scriptPath = Path.GetFullPath(Path.Combine(Application.dataPath, "../../main.py"));
+
+        // Check if the python environment and the main file of the python script exist
+        if (!File.Exists(pythonPath))
+        {
+            UnityEngine.Debug.LogError("Python environment not found at: " + pythonPath);
+            return;
+        }
+
+        if (!File.Exists(scriptPath))
+        {
+            UnityEngine.Debug.LogError("Python script not found at: " + scriptPath);
+            return;
+        }
+
+        // Create a string with all of the arguments
+        string arguments = " --incline_angle " + inclineAngle;
```

Comment:
Maybe add a todo to OOP this part, gonna be quite annoying once we have more in but for demo doesn't matter

Reply:
yea I agree later